File: projekat-eo/src/main/java/com/spring/projekateo/model/Payment.java
package com.spring.projekateo.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "payment")
public class Payment {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="payment_id",unique = true, nullable = false)
	private Integer id;
	
	@Column(name = "date", nullable = false)
	private Date date;
	
	@Column(name = "amount", nullable = false)
	private double amount;
	
	@Column(name = "purpose")
	private String purpose;

	@ManyToOne
	@JoinColumn(name = "account_id", referencedColumnName = "account_id", nullable = false)
	private Account account;
	
	public Payment() {
		
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Date getDate() {
		return date;
	}

	public void setDate(Date date) {
		this.date = date;
	}

	public double getAmount() {
		return amount;
	}

	public void setAmount(double amount) {
		this.amount = amount;
	}

	public String getPurpose() {
		return purpose;
	}

	public void setPurpose(String purpose) {
		this.purpose = purpose;
	}

	public Account getAccount() {
		return account;
	}

	public void setAccount(Account account) {
		this.account = account;
	}

}

File: projekat-eo-angular/angularApp/src/app/model/payment.ts
import { Account } from "./account";

export class Payment implements PaymentInterface{
	public id: number;
	public date: Date;
    public amount: number;
    public purpose : string;
	public account: Account;
		
	constructor(paymentCfg: PaymentInterface)
	{	
		this.id = paymentCfg.id;
		this.date = paymentCfg.date;
        this.amount = paymentCfg.amount;
        this.purpose = paymentCfg.purpose;
		this.account = paymentCfg.account;
	}
}

interface PaymentInterface {
	id?: number;
	date: Date;
    amount: number;
    purpose : string;
	account: Account;
}